compiler/compiler.py

```diff
@@ -25,11 +28,17 @@ def get_top_loop_switch(self):
 class Compiler(NodeVisitor):
     def __init__(self, parser):
         self.parser = parser
+        self._symboltable = None
         self.stack_pos = 0  # This will become an object later
+        self.scope_stack = ScopeStack()
         self.loop_switch_stack = LoopSwitchStack()
 
+
     def compile(self):
         head = self.parser.parse()
+        self.symbol_builder = SymbolTableBuilder()
+        self.symbol_builder.visit(head)
```

Comment:
symbol_builder should be a method variable not an instance variable because we don't need to keep it
symbol_builder = SymbolTableBuilder()